service/src/main/java/uk/gov/hmcts/reform/fpl/service/validators/EventChecker.java

```diff
@@ -0,0 +1,104 @@
+package uk.gov.hmcts.reform.fpl.service.validators;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.fpl.enums.Event;
+import uk.gov.hmcts.reform.fpl.model.CaseData;
+
+import java.util.EnumMap;
+import java.util.List;
+import javax.annotation.PostConstruct;
+
+import static java.util.Collections.emptyList;
+import static java.util.Optional.ofNullable;
+import static uk.gov.hmcts.reform.fpl.enums.Event.ALLOCATION_PROPOSAL;
+import static uk.gov.hmcts.reform.fpl.enums.Event.APPLICANT;
+import static uk.gov.hmcts.reform.fpl.enums.Event.CASE_NAME;
+import static uk.gov.hmcts.reform.fpl.enums.Event.CHILDREN;
+import static uk.gov.hmcts.reform.fpl.enums.Event.DOCUMENTS;
+import static uk.gov.hmcts.reform.fpl.enums.Event.FACTORS_AFFECTING_PARENTING;
+import static uk.gov.hmcts.reform.fpl.enums.Event.GROUNDS;
+import static uk.gov.hmcts.reform.fpl.enums.Event.HEARING_NEEDED;
+import static uk.gov.hmcts.reform.fpl.enums.Event.ORDERS_NEEDED;
+import static uk.gov.hmcts.reform.fpl.enums.Event.RESPONDENTS;
+import static uk.gov.hmcts.reform.fpl.enums.Event.RISK_AND_HARM;
+import static uk.gov.hmcts.reform.fpl.enums.Event.SUBMIT_APPLICATION;
+
+@Service
+public class EventChecker {
+
+    @Autowired
+    private CaseNameValidator caseNameValidator;
```

Comment:
All of these could probably made final and initialised in a constuctor that can be hidden with `@RequiredArgsConstructor(onConstructor_ = {@Autowired})`

Reply:
2 reasons why i did it like this,
- dont want to have 15args constructor (in DI env it does not matter a lot)
- there is circular dependency (EventChecker -> CaseSubmissionGuard -> EventChecker) which wont work with constructor injection

Reply:
Ahh missed the circular dependancy